connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -387,4 +402,50 @@ boolean isHeartbeatTopic(String topic) {
     String formatRemoteTopic(String topic) {
         return replicationPolicy.formatRemoteTopic(sourceAndTarget.source(), topic);
     }
+
+    // Writing the translated offsets to the __consumer_offsets topic in the destination cluster,
+    // as long as no consumers in that group are connected to the destination cluster
+    private void syncCommitOffset() throws InterruptedException, ExecutionException {
+        MirrorClient mirrorClient = new MirrorClient(config.targetConsumerConfig());
+        Map<String, Map<TopicPartition, OffsetAndMetadata>> allOffsets =
+            mirrorClient.remoteConsumerOffsets(config.sourceClusterAlias(), config.adminTimeout());
+
+        // track the consumer that are not actively consuming from the replicated topic with same consumer group id
+        Set<String> idleConsumers = new HashSet<>();
+        DescribeConsumerGroupsResult describedGroups = targetAdminClient.describeConsumerGroups(allOffsets.keySet());
```

Comment:
Should we split groups across Tasks instead of syncing all groups in a Connector? For example, MirrorCheckpointConnector divides all groups among MirrorCheckpointTasks, and each Task generates checkpoints for its assigned groups only. This is potentially more scalable.

Reply:
thanks, that sounds a scalable way, will attempt this change in the next iteration